components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41.java

```diff
@@ -453,6 +453,15 @@ public Response addServiceGroup(
             if (e.getCause().getMessage().contains("already exists")) {
                 return Response.status(Response.Status.CONFLICT).entity(new StatusResponseBean(
                         Response.Status.CONFLICT.getStatusCode(), "Cartridge group not found")).build();
+            } else if(e.getCause().getMessage().contains("duplicate cartridges")) {
+                return Response.status(Response.Status.BAD_REQUEST).entity(new StatusResponseBean(
+                        Response.Status.BAD_REQUEST.getStatusCode(), "Cartridges duplicated in the group definition")).build();
+            } else if(e.getCause().getMessage().contains("duplicate groups")) {
+                return Response.status(Response.Status.BAD_REQUEST).entity(new StatusResponseBean(
+                        Response.Status.BAD_REQUEST.getStatusCode(), "Groups duplicated in the group definition")).build();
+            } else if(e.getCause().getMessage().contains("cyclic group")) {
+                return Response.status(Response.Status.BAD_REQUEST).entity(new StatusResponseBean(
+                        Response.Status.BAD_REQUEST.getStatusCode(), "Cyclic group behaviour identified in the group definition")).build();
```

Comment:
IMO e.getCause().getMessage().contains("") is the not a good way to filter out the Status.Can't we return the status code with the exception it self?